core/src/kura.rs

```diff
@@ -362,15 +348,30 @@ impl Unlocked {
 
     /// Try to acquire lockfile and convert self to [`Locked`]
     fn lock(self) -> Result<Locked> {
+        let path = self.join(LOCK_FILE_NAME);
         if let Err(e) = fs::File::options()
             .read(true)
             .write(true)
             .create_new(true)
-            .open(self.join(LOCK_FILE_NAME))
+            .open(path.clone())
         {
             match e.kind() {
                 std::io::ErrorKind::AlreadyExists => Err(Error::Locked(self.0)),
-                _ => Err(e.into()),
+                std::io::ErrorKind::NotFound => {
+                    std::fs::create_dir_all(self.0.clone())
+                        .map_err(|e| Error::IO(e, path.clone()))?;
+                    if let Err(e) = fs::File::options()
+                        .read(true)
+                        .write(true)
+                        .create_new(true)
+                        .open(path.clone())
+                    {
+                        Err(Error::IO(e, path))
+                    } else {
+                        Ok(Locked(self.0))
+                    }
+                }
```

Comment:
Do we need this error handling branch for block store directory creation?
Since then when `lock` function is called block store directory should already be created by `create_files_if_they_do_not_exist`  function.

Reply:
That's actually why I needed to create this PR. It doesn't.

Reply:
Oh got it now, we create block folder store on already `locked` store (so `.create_files_if_they_do_not_exist()` in reality called after `.lock()`).
I'm still a bit concerned that we do the same thing twice.

Reply:
> I'm still a bit concerned that we do the same thing twice.

These things are accidentally the same. In reality there will be other files all of which need to be handled. I agree that maybe factoring it out into a function might be good, but given that it's just a sequence of operations, I don't necessarily think it'd help readability as much as obscure the intention.